From 25e75d62b25c930e744dcd92faf4f16f821b9b36 Mon Sep 17 00:00:00 2001
From: Patrick von Reth <vonreth@kde.org>
Date: Wed, 12 Jun 2013 13:11:18 +0200
Subject: [PATCH] KDE doesnt know the windows specific mimetype aplications,
 displaying the open with dialog for every file is
 impractible, so use QDesktopServices::openUrl to let
 windows decide how to open the file

---
 kio/kio/krun.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/kio/kio/krun.cpp b/kio/kio/krun.cpp
index cd20bb5..ad5656e 100644
--- a/kio/kio/krun.cpp
+++ b/kio/kio/krun.cpp
@@ -80,6 +80,8 @@
 
 #ifdef Q_WS_X11
 #include <kwindowsystem.h>
+#elif defined(Q_WS_WIN)
+#include <QDesktopServices>
 #endif
 
 KRun::KRunPrivate::KRunPrivate(KRun *parent)
@@ -173,10 +175,16 @@ bool KRun::runUrl(const KUrl& u, const QString& _mimetype, QWidget* window, bool
     KService::Ptr offer = KMimeTypeTrader::self()->preferredService(_mimetype);
 
     if (!offer) {
+#ifdef Q_WS_WIN
+        // As KDE on windows doesnt know about the windows default applications offers will be empty in nearly all cases.
+        // So we use QDesktopServices::openUrl to let windows decide how to open the file
+        return QDesktopServices::openUrl(u);
+#else
         // Open-with dialog
         // TODO : pass the mimetype as a parameter, to show it (comment field) in the dialog !
         // Hmm, in fact KOpenWithDialog::setServiceType already guesses the mimetype from the first URL of the list...
         return displayOpenWithDialog(lst, window, tempFile, suggestedFileName, asn);
+#endif
     }
 
     return KRun::run(*offer, lst, window, tempFile, suggestedFileName, asn);
-- 
1.8.0.msysgit.0

